examples/simple.rs

```diff
@@ -32,7 +32,10 @@ fn main() {
                 .register_task(task!(website_index))
                 .register_task(task!(website_about)),
         )
-        .execute();
+        .execute()?
+        .display_stats();
```

Comment:
Is there also a way to return the stats and not just display them?

Reply:
Currently you can either return the stats, or display the stats. It would be trivial to add a wrapper to make a third option of returning and displaying the stats both, I just don't have a use case. 

For now I chose to simply consume the stats (and errors) when displaying them, as otherwise you end up needing to do something like this:
```
let _ = GooseAttack::...
```
I may rename `display_stats()` to just `display()` as this may also be where we display errors in the normal CLI use-case. I will explore what's required for that shortly.

Reply:
Just .display() feels nice — it would be cool if that optionally could take the stats so that you can do:

let goose_atk = ...;
let stats = goose_atk.get_stats();
goose_atk.display(Some(stats));

Or so ...

Reply:
Errors are all handled through the logger already, so it turns out the only thing we have to display are stats. But sure, `.display()` still works.

I'm not sure what you're trying to accomplish with your example? Currently you could do that, but you'd have to pass in `goose_atk` in your example:

```rust
let goose_atk = ...
// do other stuff
display_stats(goose_atk);
```

Reply:
What I mean is:

I thought it might be possible to return the stats, modify them (custom cleanup) and then display them.

Reply:
I dislike APIs that encapsulate data needlessly - I like that it's easy to display stats though.

Reply:
The issue here is that this is all internal state, and the stats are not likely to be of use without some of this internal state (ie, information like when the test started, how long it ran...)

I suggest we create an issue and tackle exposing the stats in a more useful way through an API (and that may be the same as your example: `let stats = get_stats(internal_goose_state)`). That said, if we remove everything but the stats themselves, we'd need a different `.display()` function that prints stats with less information at its disposal.